the-collab-lab/tcl-34-smart-shopping-list
# Check duplicated items

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

We include a function:
* ```normalizer```:  this function receives an input and returns the same input in uppercase and eliminates some punctuation marks  ```. , ; :```.

We iterate over the React state to compare using  ```.find```to search for duplicates. We as well normalized the state without modifying it and the user's input.
We fixed two bugs:
* ```setNewItem```: We eliminated this callback function was not declared by the father component
* ```setMessage```: This function evaluated if there were any error using ```error ? some message : some other message```, we replaced the ```error``` with  ```checkProduct```which is the function that checks duplicates.

## Related Issue
Closes #6 

## Acceptance Criteria

1. Mostrar un mensaje de error si el usuario intenta enviar un nuevo elemento que tiene exactamente el mismo nombre que un elemento existente.
2. Mostrar un mensaje de error si el usuario intenta enviar un nuevo elemento que tiene el mismo nombre que un elemento existente, en el que se han normalizado las mayúsculas y se han eliminado los signos de puntuación.
3. La entrada original del usuario debe ser la que se guarde en la base de datos

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓ | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### After
![image](https://user-images.githubusercontent.com/42523734/138172781-341c26a0-0a91-4192-abfc-c3f198077caf.png)


## Testing Steps / QA Criteria

1. From your terminal, pull down this branch with ```git pull origin sr-yu-check-duplicates``` and check that branch out with ```git checkout sr-yu-check-duplicates```.
2. Then ```npm start``` to launch the app.
3. Try adding a new item to the list and then you click again to see how the message changes.
